File: 08-Greedy-Algorithm/python/02_minimum_number_of_platforms.py
class Solution:
    def findPlatform(self, Arrival: List[int], Departure: List[int]) -> int:
        Arrival.sort()
        Departure.sort()

        n = len(Arrival)
        platforms, max_platforms = 1, 1
        i, j = 1, 0

        while i < n and j < n:
            if Arrival[i] <= Departure[j]:
                platforms += 1
                i += 1
            else:
                platforms -= 1
                j += 1
            max_platforms = max(max_platforms, platforms)

        return max_platforms

File: 08-Greedy-Algorithm/cpp/02-minimum-number-of-platforms.cpp
class Solution {
public:
    int findPlatform(vector<int>& Arrival, vector<int>& Departure) {
        sort(Arrival.begin(), Arrival.end());
        sort(Departure.begin(), Departure.end());

        int n = Arrival.size();
        int platforms = 1, maxPlatforms = 1;
        int i = 1, j = 0;

        while (i < n && j < n) {
            if (Arrival[i] <= Departure[j]) {
                platforms++;
                i++;
            } else {
                platforms--;
                j++;
            }
            maxPlatforms = max(maxPlatforms, platforms);
        }

        return maxPlatforms;
    }
};
